sympy/polys/polytools.py

```diff
@@ -7938,3 +7937,339 @@ def named_poly(n, f, K, name, x, polys):
+    """ Helper function to combine factored polynomial equations into minimal factor systems.
+
+    Takes lists of (factor, conditions) pairs and returns minimal combinations that
+    satisfy the system of equations.
+    """
+
+    factor_dict = defaultdict(list)
+    for eq in factored_eqs:
+        for factor, conditions in eq:
+            factor_dict[factor].extend(conditions)
+
+    eq_factors = [{factor for factor, _ in eq} for eq in factored_eqs]
+
+    all_combinations = _find_minimal_factor_combinations(eq_factors)
+
+    minimal_combinations = []
+    for combo in all_combinations:
+        if not any(combo > other for other in all_combinations):
+            minimal_combinations.append(combo)
+
+    return [[(factor, factor_dict[factor]) for factor in combo]
+            for combo in minimal_combinations]
+
+
+def factor_system_cond(eqs, gens=None, **kwargs):
+    """Factor a system of equations and return factors with symbolic conditions.
+
+    Converts equations to polynomials, factors them, and returns factors with
+    Boolean conditions expressing when each factor is valid. Handles both
+    polynomial and transcendental equations.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of equations or expressions to be factored. Can include:
+        - Polynomial equations
+        - Transcendental equations
+        - Zero expressions
+        - Constants
+    gens : list, optional
+        List of generators (variables). If None, determined automatically
+        from the equations' free symbols
+    **kwargs : dict
+        Additional keyword arguments for factorization routines
+
+    Returns
+    =======
+
+    list of list of tuple
+        Each inner list represents a solution set.
+        Each tuple contains (factor, condition) where:
+        - factor is a symbolic expression (not a Poly)
+        - condition is either:
+            * sympy.true if no conditions apply
+            * Or(Ne(c1, 0), Ne(c2, 0), ...) for non-zero conditions
+
+    Examples
+    ========
+
+    >>> from sympy import symbols
+    >>> from sympy.polys.polytools import factor_system_cond
+    >>> x, a, b = symbols('x a b')
+    >>> factor_system_cond([(a - 1)*(x - 2), (b - 3)*(x - 2)], [x])
+    [[(x - 2, Ne(a - 1, 0) | Ne(b - 3, 0))]]
+
+    Notes
+    =====
+
+    - Returns [] if input list is empty
+    - Returns [[]] if all expressions are zero, indicating infinite solutions
+    - Returns [] if any non-zero constant is present, indicating no solution exists
+    - Handles transcendental equations that can be converted to polynomial form
+    - Maintains symbolic conditions for factor validity
+    """
+
+    if not eqs:
+        return []
+
+    exprs = [sympify(eq) for eq in eqs]
+    exprs = [expr.simplify() for expr in exprs]
+
+    if all(expr.is_zero for expr in exprs):
+        return [[]]
+
+    if any(expr.is_constant() and not expr.is_zero for expr in exprs):
+        return []
```

Comment:
The `is_constant` function is poorly defined and should be avoided. We can check for constants after converting to `Poly` when it becomes more well defined.